collect_app/src/main/java/org/odk/collect/android/fragments/ShowQRCodeFragment.java

```diff
@@ -236,17 +236,25 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
             if (resultCode == Activity.RESULT_OK) {
                 try {
                     final Uri imageUri = data.getData();
-                    final InputStream imageStream = getActivity().getContentResolver()
-                            .openInputStream(imageUri);
-
-                    final Bitmap bitmap = BitmapFactory.decodeStream(imageStream);
-                    String response = QRCodeUtils.decodeFromBitmap(bitmap);
-                    if (response != null) {
-                        applySettings(response);
+                    if (imageUri != null) {
+                        final InputStream imageStream = getActivity().getContentResolver()
+                                .openInputStream(imageUri);
+
+                        final Bitmap bitmap = BitmapFactory.decodeStream(imageStream);
+                        if (bitmap != null) {
```

Comment:
We should also display a toast message if this condition is not met to notify the user that they need to select the QRCode again

Reply:
Fixed.